src/DataMiner/SLNetTypes/Automation/AutomationDownloadButtonOptions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Skyline.DataMiner.Automation;
+
+namespace SLNetTypes.Automation
+{
+    /// <summary>
+    /// This class allows you to create a download button in an interactive Automation script.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// var downloadButtonOptions = new AutomationDownloadButtonOptions()
+    /// {
+    ///     URL = @"/Documents/DMA_COMMON_DOCUMENTS/DailyReport.pdf",
+    ///     StartDownloadImmediately = false
+    ///     ReturnWhenDownloadIsStarted = false,
+    ///     FileNameToSave = "Report.PDF",
+    /// };
+    /// UIBlockDefinition blockItem = new UIBlockDefinition
+    /// {
+    ///     Type = UIBlockType.DownloadButton,
+    ///     Width = 125,
+    ///     Text = "Get report of today",
+    ///     ConfigOptions = downloadButtonOptions,
+    /// };
+    /// uiBuilder.AppendBlock(blockItem);
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// <note type="note">
+    /// <para>If the name of a variable starts with the following prefix, IntelliSense in DataMiner Cube will list the object properties: downloadButtonConfig, downloadButtonOptions*</para>
+    /// </note>
+    /// </remarks>
+    public class AutomationDownloadButtonOptions : AutomationDateTimeUpDownOptions
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether the download will start immediately when the component is displayed.
+        /// </summary>
+        /// <value><c>true</c> to immediately start the download without clicking the button; otherwise, <c>false</c>.</value>
+        /// <remarks>
+        /// <para>Default: <c>false</c>.</para>
+        /// </remarks>
+        public bool StartDownloadImmediately { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating which file must be downloaded. This URL can be either absolute path or relative path.
+        /// When using an absolute URL, then it is recommended that the file is public accesible from the web. 
```

Comment:
I would remove the lines on line nrs 45 and 46. I think they only add confusion.